src/net.h

```diff
@@ -691,32 +689,60 @@ class CNode
     uint256 hashContinue;
     std::atomic<int> nStartingHeight{-1};
 
-    // flood relay
-    std::vector<CAddress> vAddrToSend;
-    CRollingBloomFilter addrKnown;
-    bool fGetAddr{false};
-    std::set<uint256> setKnown;
-    int64_t nNextAddrSend GUARDED_BY(cs_sendProcessing){0};
-    int64_t nNextLocalAddrSend GUARDED_BY(cs_sendProcessing){0};
-
-    // inventory based relay
-    CRollingBloomFilter filterInventoryKnown GUARDED_BY(cs_inventory);
-    // Set of transaction ids we still have to announce.
-    // They are sorted by the mempool before relay, so the order is not important.
-    std::set<uint256> setInventoryTxToSend;
+    struct AddrRelay {
+        CCriticalSection cs_addrsend;
+        std::vector<CAddress> vAddrToSend GUARDED_BY(cs_addrsend);
+        CRollingBloomFilter addrKnown GUARDED_BY(cs_addrsend);
+        bool fGetAddr GUARDED_BY(cs_addrsend){false};
+        int64_t nNextAddrSend GUARDED_BY(cs_addrsend){0};
+        int64_t nNextLocalAddrSend GUARDED_BY(cs_addrsend){0};
+
+        AddrRelay() : addrKnown(5000, 0.001) {}
+    };
+
+    AddrRelay m_addr_relay;
+    const bool m_addr_relay_peer;
```

Comment:
nit: `m_addr_relay_peer` could be private

Reply:
Meh, going to leave this as-is.